detox/src/realms/DetoxPrimaryContext.test.js

```diff
@@ -186,23 +197,23 @@ describe('DetoxPrimaryContext', () => {
       await facade.init();
 
       expect(process.env.DETOX_CONFIG_SNAPSHOT_PATH).toBeDefined();
-      expect(process.env.DETOX_CONFIG_SNAPSHOT_PATH).toMatch(tempFileRegexp);
+      expect(process.env.DETOX_CONFIG_SNAPSHOT_PATH).toMatch(TEMP_FILE_REGEXP);
     });
 
     it('should install a worker if worker ID has been specified', async () => {
-      await facade.init({ workerId });
-      expect(facade.session).toEqual(expect.objectContaining({ workerId }));
+      await facade.init({ workerId: WORKER_ID });
+      expect(facade.session).toEqual(expect.objectContaining({ workerId: WORKER_ID }));
       expect(detoxWorker().init).toHaveBeenCalled();
     });
 
     it('should register the worker at the IPC server\'s', async () => {
-      await facade.init({ workerId });
-      expect(ipcServer.onRegisterWorker).toHaveBeenCalledWith({ workerId });
+      await facade.init({ workerId: WORKER_ID });
+      expect(ipcServer().onRegisterWorker).toHaveBeenCalledWith({ workerId: WORKER_ID });
     });
 
     describe('given an initialization failure', () => {
       it('should report status as "init"', async () => {
-        ipcServer.init.mockImplementation(() => throwErrorImpl('init error'));
+        IPCServer.prototype.init = jest.fn().mockRejectedValue(new Error('init failed'));
```

Comment:
I checked what you tried to achieve with that complex mocking (4 lines) for that IPCServer, and realized you could just mock the prototype, which is pretty idiomatic for JS developers.

Reply:
Right, thanks!